kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/BigQueryWriter.java

```diff
@@ -145,6 +163,35 @@ public void writeRows(PartitionedTableId table,
         mostRecentException);
   }
 
+  /**
+   * Decide whether the failure is a partial failure or complete failure
+   * @param numOfFailedRows Number of rows that failed
+   * @param numOfAllRows Number of rows to insert
+   * @return isPartialFailure
+   */
+  private boolean isPartialFailure(int numOfFailedRows, int numOfAllRows) {
+    return numOfFailedRows == numOfAllRows;
+  }
+
+  /**
+   * Filter out succeed rows, and return a list of failed rows.
+   * @param rows The rows to write.
+   * @param failRowsIndex A set of failed row index
+   * @return A list of failed rows
+   */
+  private List<InsertAllRequest.RowToInsert> getFailedRows(List<InsertAllRequest.RowToInsert> rows,
+                                                           Set<Long> failRowsIndex) {
+    logger.info("{} rows succeeded, {} rows failed", rows.size() - failRowsIndex.size(), failRowsIndex.size());
```

Comment:
I think this logging message should be in writeRows itself.

Reply:
Sure, my previous concern if we log somewhere else, we have to iterate through the list again, which might be expensive. While partial failure is not very common, I think you are right.